### ISSUE TITLE
getDevices() and switchDevice() is not working in some devices using v2.6.1

### ISSUE THREAD

**Issue by monumehta** [AgoraIO/Basic-Video-Broadcasting#15]

**Comment by shubhamparamhans** [comment]
I am getting DOM exception from agora js in few devices.
I haven't found any pattern yet though.

**Comment by plutoless** [comment]
@monumehta is there any error logs for reference?

**Comment by plutoless** [comment]
@shubhamparamhans could you pls paste the dom exception logs so that we can analyze?

**Comment by shubhamparamhans** [comment]
![agora-1](https://user-images.githubusercontent.com/982056/57178719-69c04680-6e92-11e9-9c8e-6976a9152295.png)

This is the error i see in console, when i use switchDevice.
Interesting thing is that i happens for few devices.
I checked the length of getDevices it was two but somehow on switching this DOM exception was visible in console

**Comment by plutoless** [comment]
@shubhamparamhans what kind of device are you getting these issues? Could you pls share me some device names which i can use to reproduce?

**Comment by shubhamparamhans** [comment]
@plutoless redmi note 4, moto e4 plus, honor 6s
On these devices switch camera was not working.

**Comment by plutoless** [comment]
@shubhamparamhans yes you are right we have some issues for some devices you mentioned above. The reason is those devices do not allow openning front/back camera at the same time. The exception is basically complaining that.
We have realized the issue and is arranging a fix in next major version but this takes some time. If you want to work around this issue you may first close 1st camera's local stream, and then create the second camera stream to achieve similar experience
